lib/reservation_tracker.rb

```diff
@@ -0,0 +1,48 @@
+# handle availabilty 
+# handle rooms
+# room availability
+module Hotel 
+ 
+    class ReservationTracker
+        attr_reader :reservations, :all_rooms, :all_availabile, :block_reservation
+
+        def initialize(max_rooms)
+            if max_rooms > 20 
+                raise ArgumentError, "this hotel is too large, max 20 rooms"
+            else 
+                @all_rooms = [*1..max_rooms]
```

Comment:
Nice, but is there a requirement to handle a changing number of max rooms? ;) I think that the requirements say that there will always be 20 rooms